front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts

```diff
@@ -49,7 +49,7 @@ async function handler(
     });
   }
 
-  if (vault.isSystem() && !auth.isAdmin()) {
+  if (vault.isSystem() && !vault.canAdministrate(auth)) {
```

Comment:
do you need both checks ?

Reply:
The rule is that for the system vault no one can write, so we use the verb `administrate`. Where `administrate` means:
- Connecting managed data sources
- Administrating vaults (name, members, ...)